/*
 * Copyright (C) 2021 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package android.hardware.automotive.vehicle.proto;

import "android/hardware/automotive/vehicle/VehiclePropertyStatus.proto";

/* Must be in sync with VehiclePropValue.aidl. */
message VehiclePropValue {
    /* Time is elapsed nanoseconds since boot */
    int64 timestamp = 1;

    /* Area type(s) for non-global property it must be one of the value from
     * VehicleArea* enums or 0 for global properties. */
    int32 area_id = 2;

    /* Property identifier */
    int32 prop = 3;

    /* Status of the property */
    VehiclePropertyStatus status = 4;

    /* This is used for properties of types VehiclePropertyType#INT
     * and VehiclePropertyType#INT_VEC */
    repeated int32 int32_values = 5;

    /* This is used for properties of types VehiclePropertyType#FLOAT
     * and VehiclePropertyType#FLOAT_VEC */
    repeated float float_values = 6;

    /* This is used for properties of type VehiclePropertyType#INT64 */
    repeated int64 int64_values = 7;

    /* This is used for properties of type VehiclePropertyType#BYTES */
    bytes byte_values = 8;

    /* This is used for properties of type VehiclePropertyType#STRING */
    string string_value = 9;
};
